tests/integration/publish_tests.rs

```diff
@@ -260,6 +297,35 @@ fn ignores_directories() {
   assert_not_contains!(output, "ignored.ts");
 }
 
+#[test]
+fn includes_directories_with_gitignore() {
+  let context = publish_context_builder().build();
+  let temp_dir = context.temp_dir().path();
+  temp_dir.join("deno.json").write_json(&json!({
+    "name": "@foo/bar",
+    "version": "1.0.0",
+    "exports": "./main.ts",
+    "publish": {
+      "include": [ "deno.json", "main.ts" ]
+    }
+  }));
+
+  temp_dir.join(".gitignore").write("main.ts");
+  temp_dir.join("main.ts").write("");
+  temp_dir.join("ignored.ts").write("");
+
+  let output = context
+    .new_command()
+    .arg("publish")
+    .arg("--token")
+    .arg("sadfasdf")
+    .run();
+  output.assert_exit_code(0);
+  let output = output.combined_output();
+  assert_contains!(output, "main.ts");
+  assert_not_contains!(output, "ignored.ts");
```

Comment:
I think maybe `main.ts` is only in the output because it's an export, but it seems like this functionality doesn't work https://github.com/denoland/deno/issues/22656